src/analyses/variable-sensitivity/constant_pointer_abstract_object.cpp

```diff
@@ -70,6 +71,35 @@ constant_pointer_abstract_objectt::merge(abstract_object_pointert other) const
   }
 }
 
+bool constant_pointer_abstract_objectt::same_target(
+  abstract_object_pointert other) const
+{
+  auto cast_other =
+    std::dynamic_pointer_cast<const constant_pointer_abstract_objectt>(other);
+
+  if(value_stack.is_top_value() || cast_other->value_stack.is_top_value())
+    return false;
+
+  bool matching_pointer = value_stack.target_expression() ==
+                          cast_other->value_stack.target_expression();
+
+  return matching_pointer;
+}
+
+exprt constant_pointer_abstract_objectt::offset_from(
+  abstract_object_pointert other) const
+{
+  auto cast_other =
+    std::dynamic_pointer_cast<const constant_pointer_abstract_objectt>(other);
+
+  if(value_stack.is_top_value() || cast_other->value_stack.is_top_value())
+    return nil_exprt();
```

Comment:
I would really like to get rid of the use of `nil_exprt()` as it feels very kludged.  Can this method return an `abstract_objectt` so it has a way of returning top.  Failing that?  `optionalt`?

Reply:
This is calling site is `return environment.eval(offset_from(rhs), ns);` so the `nil_exprt()` does end up as a TOP value.